publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/identifiers/HandleIdentifierEventHandler.java

```diff
@@ -122,21 +123,25 @@ protected Void processInputPayload(EventReference input) {
         var eventBlob = s3Driver.readEvent(input.getUri());
 
         if (RESOURCE_UPDATE_EVENT_TOPIC.equals(input.getTopic())) {
-            var resourceUpdate = parseResourceUpdateInput(eventBlob);
-            if (isPublished(resourceUpdate) && isMissingHandle(resourceUpdate)) {
-                logger.info("Creating handle for publication: {}", resourceUpdate.getIdentifier());
-                var publication = fetchPublication(resourceUpdate.getIdentifier());
-                var additionalIdentifiers = new HashSet<>(publication.getAdditionalIdentifiers());
-                var handle = createNewHandle(getLandingPage(publication.getIdentifier()));
-                logger.info("Created handle: {}", handle.value());
-                additionalIdentifiers.add(handle);
-                publication.setAdditionalIdentifiers(additionalIdentifiers);
-                attempt(() -> resourceService.updatePublication(publication));
-            }
+            parseResourceUpdateInput(eventBlob)
+                    .filter(HandleIdentifierEventHandler::isPublished)
+                    .filter(HandleIdentifierEventHandler::isMissingHandle)
+                    .ifPresent(this::createAndAssignHandle);
         }
         return null;
     }
 
+    private void createAndAssignHandle(Resource resourceUpdate) {
+        logger.info("Creating handle for publication: {}", resourceUpdate.getIdentifier());
+        var publication = fetchPublication(resourceUpdate.getIdentifier());
+        var additionalIdentifiers = new HashSet<>(publication.getAdditionalIdentifiers());
+        var handle = createNewHandle(getLandingPage(publication.getIdentifier()));
+        logger.info("Created handle: {}", handle.value());
+        additionalIdentifiers.add(handle);
+        publication.setAdditionalIdentifiers(additionalIdentifiers);
+        attempt(() -> resourceService.updatePublication(publication));
+    }
+
```

Comment:
This is the old logic unchanged, just extracted to a helper method because I had to add a third condition and it was getting silly.